src/components/CreditCardForm/CreditCardForm.tsx

```diff
@@ -0,0 +1,45 @@
+import './style.css'
+import { useRef, ChangeEvent, useEffect } from 'react'
+
+export const CreditCardForm = (): JSX.Element => {
+  const inputRef1 = useRef<HTMLInputElement>(null!)
```

Comment:
Dvě možné vylepšení:

**Vykřičníky**
Vykřičník sice používat můžeš, ale může to vést k nepředvídatelným chybám - říkáš, že si jsi naprosto jistá tím `null`, které `useRef` hooku předáváš. Kdyby tam byl nějaký error, už tě na to VSCode neupozorní, protože `!` má vždycky pravdu. Mnohem jednodušší by bylo to otypovat ne jako `<HTMLInputElement>`, ale `<HTMLInputElement | null>`.

**Pojmenování čísly**
Pojmenování skrze čísla by ti tady asi neprošlo přes code review. Čísla se můžou změnit, jakkoli si myslíme že ne. Třeba v jednom systému, který potřeboval validaci rodných čísel měli u nás v Česku jeden známý bug - nepočítali s tím, že před určitým rokem rodná čísla vypadala jinak. I počet inputů tady se může změnit nebo někdo může chtít, abys tenhle komponent použila pro jistý účel. A co když bude potřeba přidat input doprostřed? Budeš mít `1, 2, 5, 3, 4`. 😂  Ale chápu, že tady je zrovna dost těžké vymyslet alternativní pojmenování.

A nebo úplně alternativně - `useRef()` může být i array `useRef([])` . 😉 Nenašla jsem moc pěkných vysvětlení ale něco jako [tady](https://adics.hashnode.dev/using-useref-to-multiple-elements-in-react-js) nebo [tady](https://clubmate.fi/set-multiple-refs).

Reply:
děkuji moc, zkusím použít něco z toho :)